src/App.vue

```diff
@@ -25,6 +26,44 @@ export default {
       // set the account globally, to reduce getAccounts calls in components
       window.account = accounts[0];
 
+      // Check if address is EOA, create localStorage to store assets on
+      let bytecode = await web3.eth.getCode(accounts[0]);
+  
+      if (bytecode === '0x') {
+        if (localStorage.getItem("receivedAssets") === null) {
+          localStorage.setItem('receivedAssets', JSON.stringify({"value":[], "account": accounts[0]}));
+        }
+        else{
+          const localStorageOwner = JSON.parse(localStorage.getItem("receivedAssets"));
+          
+          if(localStorageOwner.account !== accounts[0]){
+            localStorage.removeItem("receivedAssets");
+          }
+        }
+        
+        if (localStorage.getItem("issuedAssets") === null){
+          localStorage.setItem('issuedAssets', JSON.stringify({"value":[], "account": accounts[0]}));
+        }
+        else{
+          const localStorageOwner = JSON.parse(localStorage.getItem("issuedAssets"));
+          
+          if(localStorageOwner.account !== accounts[0]){
+            localStorage.removeItem("issuedAssets");
+          }
```

Comment:
I think we can clean up this conditional block. The logic for `receivedAssets` and `issuedAssets` is the same so you could create a function e.g `setupLocalStorage` and pass in the local storage key you want to use.

Then this becomes 
```js
if (bytecode) {
  setupLocalStorange('receivedAssets')
  setupLocalStorange('issuedAssets')
} else {
  ...
}
```